### ISSUE TITLE
Ability to Export & Import data cross platform

### ISSUE THREAD

**Issue by jittygitty** [session-foundation/session-android#216]
**Describe the bug**

I tried to move the Desktop app to another computer, good luck, didn't even easily see where the data is stored. Such app must either be PORTABLE so it can be copied/moved to any other computer, and or at least "or" be able to EXPORT all your data/chats etc.

**To reproduce**

Use the app, experience a type of vendor "lock-in" with your data and even worse can't even move your data to the same app installation but on a different device.

 - Device: Windows 10 PC
 - OS: Windows 10
 - Version of Session or latest commit hash, hmm there's not even a quick easy way to see the current version of the app? 

**Additional context**

I really want to LOVE this app, in fact I do love it because it does some big things right. I really want to USE it, but I can't for the big reason that I can't have my data "stuck".
I LIKE that the data is not on a remote server like with Element.io chat for example. I like that the data is LOCAL to the device app is installed to. But what if you lose the device? How can you easily sync/backup your data? We can't even EXPORT our data right now. We currently can't take our data and move it to another installation on another device, ie portability. Add such features and many of us I think will get our  network of people/friends/family/colleagues etc to start using this app.

If somehow I'm wrong and this feature exists and I missed it, my apologies and I'd appreciate a pointer on where/how etc. Thanks!

**Comment by oceanside-chess** [comment]
Yes, I think having the ability to export message would be awesome. It's a much needed feature that the project is really missing. Every other privacy respecting messaging app I can think of has an implementation of something similar.

**Comment by ThomasSession** [comment]
> Yes, I think having the ability to export message would be awesome. It's a much needed feature that the project is really missing. Every other privacy respecting messaging app I can think of has an implementation of something similar.

Yes it's definitely something we want to add. The reason it hasn't happened yet is because we have plans for refactoring some of the underlying data structure, and an import/export make sense to happen once we have decided on and finalised those underlying changes.

**Comment by JuanLuuB** [comment]
> > Yes, I think having the ability to export message would be awesome. It's a much needed feature that the project is really missing. Every other privacy respecting messaging app I can think of has an implementation of something similar.
> 
> Yes it's definitely something we want to add. The reason it hasn't happened yet is because we have plans for refactoring some of the underlying data structure, and an import/export make sense to happen once we have decided on and finalised those underlying changes.

Hello, do you know when this feature will be available? Thanks for the app, is super epic.

**Comment by ThomasSession** [comment]
> > > Yes, I think having the ability to export message would be awesome. It's a much needed feature that the project is really missing. Every other privacy respecting messaging app I can think of has an implementation of something similar.
> > 
> > 
> > Yes it's definitely something we want to add. The reason it hasn't happened yet is because we have plans for refactoring some of the underlying data structure, and an import/export make sense to happen once we have decided on and finalised those underlying changes.
> 
> Hello, do you know when this feature will be available? Thanks for the app, is super epic.

We don't have a set timeline but it's definitely something we want to sort out.
The reason I listed above is still true now.
The data structure is different on all 3 platforms which makes things harder.
So being able to export and import data between platforms is definitely going to be tricky as things are right now.
And even within a given platform, on Android in this case, we need to make sure we have a data layer we are happy with before committing to allowing export and imports. 
It is a highly requested feature though so it's definitely something we want to support in the future.
